### PR TITLE
[Term Entry]: Py torch tensor operations .gather()

### PR DESCRIPTION
**Pull request by Radhika-okhade**
### Description

An introduction to the concept
A Syntax section that provides the syntax for the concept
An Example section that provides an example demonstrating the concept in use

### Issue Solved

#5563

### Type of Change

- Adding a new entry

### Checklist

- [✅ ] All writings are my own.
- [✅ ] My entry follows the Codecademy Docs style guide.
- [✅ ] My changes generate no new warnings.
- [✅ ] I have performed a self-review of my own writing and code.
- [ ✅] I have checked my entry and corrected any misspellings.
- [✅ ] I have made corresponding changes to the documentation if needed.
- [ ✅] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ✅] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [✅ ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/pytorch/concepts/tensor-operations/terms/gather/gather.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+Title: '.gather()'
+Description: 'Retrieves specific elements from a tensor along a defined axis, ideal for selecting values based on indices in data processing and machine learning.'
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/gather/gather.md:3]:
```suggestion
Description: 'Retrieves specific elements from a tensor along a defined axis based on indices.'
```

File: content/pytorch/concepts/tensor-operations/terms/gather/gather.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  - 'AI'
+  - 'Data Types'
+  - 'Deep Learning'
+  - 'Functions'
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'paths/data-science'
+---
+
+The **`.gather()`** function in PyTorch is a tensor operation that enables the retrieval of specific elements from a tensor along a specified axis. This function is particularly useful for selecting values based on a set of indices, making it ideal for applications in machine learning and data processing where efficient data selection is critical.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/gather/gather.md:17]:
```suggestion
The **`.gather()`** function in PyTorch is a tensor operation that retrieves specific elements from a tensor along a specified axis. It is beneficial for selecting values based on a set of indices, making it ideal for applications in machine learning and data processing, where efficient data selection is critical.

```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensor-operations/terms/gather/gather.md:17]: Modifications: reframed for better readability and also leave a line after the sentence before going to the ## Syntax section

File: content/pytorch/concepts/tensor-operations/terms/gather/gather.md (hunk 1 above)
Comment (mamtawardhani) [reply content/pytorch/concepts/tensor-operations/terms/gather/gather.md:3]: Description should be the 'job' of the function, not the uses. Uses can be mentioned later in the introduction section

File: content/pytorch/concepts/tensor-operations/terms/gather/gather.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+## Syntax
+
+```pseudo
+torch.gather(input, dim, index)
+```
+
+- `input`: The source tensor from which values are gathered.
+- `dim`: The dimension along which the gather operation is performed.
+- `index`: A tensor of indices specifying which values to gather from each row (or column) along the specified dimension.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/gather/gather.md:26]:
```suggestion
- `input`: The source tensor from which values are gathered.
- `dim`: The dimension along which to gather values. This is the axis in the `input` tensor where the selection occurs.
- `index`: A tensor of indices specifying which values to gather from the `input` tensor along the specified `dim`.
```

File: content/pytorch/concepts/tensor-operations/terms/gather/gather.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+The function returns a tensor of the same shape as `index`, where each value is gathered from the `input` tensor based on the specified `index`.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/gather/gather.md:28]:
```suggestion
The function returns a tensor with the same shape as the `index`, where each value is gathered from the `input` tensor based on the specified indices.
```

File: content/pytorch/concepts/tensors/terms/view/view.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+---
+Title: '.view()'
+Description: 'Reshapes a tensor to a specified shape without changing data, ideal for adjusting tensor dimensions to fit model requirements while preserving element count.'
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/view/view.md:3]:
```suggestion
Description: 'Reshapes a tensor to a specified shape without changing its data, as long as the total number of elements remains the same.'
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensors/terms/view/view.md:3]: same comment as the previous entry

File: content/pytorch/concepts/tensors/terms/view/view.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+## Syntax
+
+```pseudo
+tensor.view(shape)
+```
+
+- `shape`: A tuple or list defining the desired dimensions. The total number of elements must match the original tensor.
+  
+
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/view/view.md:27]: Remove the extra line

File: content/pytorch/concepts/tensor-operations/terms/gather/gather.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+# Define the indices to gather
+index_tensor = torch.tensor([[0, 1], [1, 0]])
+
+# Gather elements from the source tensor along dimension 1
+output_tensor = torch.gather(input_tensor, 1, index_tensor)
+
+print(output_tensor)
+```
+This example results in the following output:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/gather/gather.md:48]: add a one line space before this

File: content/pytorch/concepts/tensors/terms/view/view.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+The function returns a tensor with the specified shape, sharing the same data as the original tensor.
+
+## Example
+
+This example reshapes a tensor from a shape of `(2, 3)` to `(3, 2)` using `.view()`.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/view/view.md:32]:
```suggestion
This example reshapes a tensor from a shape of `(2, 3)` to `(3, 2)` using `.view()`:
```

File: content/pytorch/concepts/tensors/terms/view/view.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+import torch
+
+# Create a tensor of shape (2, 3)
+original_tensor = torch.tensor([[1, 2, 3], [4, 5, 6]])
+
+# Reshape the tensor to (3, 2)
+reshaped_tensor = original_tensor.view(3, 2)
+
+print(reshaped_tensor)
+```
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/view/view.md:44]: leave a one line space after this

File: content/pytorch/concepts/tensors/terms/view/view.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+## Output
+
+This example results in the following output:
+
+```shell
+tensor([[1, 2],
+        [3, 4],
+        [5, 6]])
+```
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/view/view.md:53]: leave a one line space after this

File: content/pytorch/concepts/tensors/terms/view/view.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+# Create a tensor of shape (2, 3)
+original_tensor = torch.tensor([[1, 2, 3], [4, 5, 6]])
+
+# Reshape the tensor to (3, 2)
+reshaped_tensor = original_tensor.view(3, 2)
+
+print(reshaped_tensor)
+```
+## Output
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/view/view.md:45]:
```suggestion
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensors/terms/view/view.md:45]: `## Output` block does not exist

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @Radhika-okhade thank you for contributing to Codecademy Docs, I've reviewed the PR and the following things should be done:

1. both the enteries are on the same branch - `[PyTorch_Tensor_Operations_.gather()]`, you will have to write the `view.md` on a different branch and raise another PR for that.
2. I've left some comments along with suggestions for the required changes, you can modify those at your earliest convenience. 😄 
Thank you!
